api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2873,7 +3394,14 @@ private String putObject(String bucketName, String objectName, int length,
       queryParamMap.put(UPLOAD_ID, uploadId);
     }
 
-    HttpResponse response = executePut(bucketName, objectName, headerMap, queryParamMap, data, length);
+    if (uploadId == null) {
```

Comment:
As this method is rest wrapper, its better to do this logic outside of this method.